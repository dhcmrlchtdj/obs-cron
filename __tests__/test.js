const cron = require("../");

jest.setTimeout(5 * 60 * 1000);

test("every minute", done => {
    let flag = 2;
    let fst = 0;
    let snd = 0;
    const sub = cron("* * * * *").subscribe(() => {
        console.log(new Date());
        if (flag === 2) {
            flag--;
            fst = new Date().getMinutes();
        } else if (flag === 1) {
            flag--;
            snd = new Date().getMinutes();

            sub.unsubscribe();

            const delta = snd - fst;
            expect(delta).toEqual(1);
            done();
        }
    });
});
