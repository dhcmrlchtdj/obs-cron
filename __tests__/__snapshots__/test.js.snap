// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exception 1`] = `"expect cron, got '* * * * * *'"`;

exports[`exception 2`] = `"expect Number, got 'x'"`;

exports[`exception 3`] = `"99 not in range [0, 59]"`;

exports[`exception 4`] = `"expect range, got '1'"`;

exports[`exception 5`] = `"1 not in range [3, 59]"`;

exports[`exception 6`] = `"expect step, got '*/'"`;

exports[`exception 7`] = `"expect step, got '1-9/'"`;

exports[`exception 8`] = `"expect step, got '1/'"`;

exports[`exception 9`] = `"expect range, got '1-'"`;

exports[`exception 10`] = `"expect list, got '1,'"`;

exports[`exception 11`] = `"expect range, got '1-2-3'"`;

exports[`exception 12`] = `"expect step, got '1/2/3'"`;

exports[`exception 13`] = `"expect list, got '1,2,'"`;
